gcloud/dns/changes.py

```diff
@@ -0,0 +1,261 @@
+
+    :type zone: :class:`gcloud.dns.zone.ManagedZone`
+    :param zone: A zone which holds one or more record sets.
+    """
+
+    def __init__(self, zone):
+        self.zone = zone
+        self._properties = {}
+        self._additions = self._deletions = ()
+
+    @classmethod
+    def from_api_repr(cls, resource, zone):
+        """Factory:  construct a change set given its API representation
+
+        :type resource: dict
+        :param resource: change set representation returned from the API
+
+        :type zone: :class:`gcloud.dns.zone.ManagedZone`
+        :param zone: A zone which holds zero or more change sets.
+
+        :rtype: :class:`gcloud.dns.changes.Changes`
+        :returns: RRS parsed from ``resource``.
+        """
+        changes = cls(zone=zone)
+        changes._set_properties(resource)
+        return changes
+
+    def _set_properties(self, resource):
+        """Helper method for :meth:`from_api_repr`, :meth:`create`, etc.
+
+        :type resource: dict
+        :param resource: change set representation returned from the API
+        """
+        resource = resource.copy()
+        self._additions = tuple([
+            ResourceRecordSet.from_api_repr(added_res, self.zone)
+            for added_res in resource.pop('additions', ())])
+        self._deletions = tuple([
+            ResourceRecordSet.from_api_repr(added_res, self.zone)
+            for added_res in resource.pop('deletions', ())])
+        self._properties = resource
+
+    @property
+    def path(self):
+        """URL path for change set APIs.
+
+        :rtype: string
+        :returns: the path based on project, zone, and change set names.
+        """
+        return '/projects/%s/managedZones/%s/changes/%s' % (
+            self.zone.project, self.zone.name, self.name)
+
+    @property
+    def name(self):
+        """Name of the change set.
+
+        :rtype: string or ``NoneType``
+        :returns: Name, as set by the back-end, or None.
+        """
+        return self._properties.get('id')
+
+    @name.setter
+    def name(self, value):
+        """Update name of the change set.
+
+        :type value: string
+        :param value: New name for the changeset.
+        """
+        if not isinstance(value, six.string_types):
+            raise ValueError("Pass a string")
+        self._properties['id'] = value
+
+    @property
+    def status(self):
+        """Status of the change set.
+
+        :rtype: string or ``NoneType``
+        :returns: Status, as set by the back-end, or None.
+        """
+        return self._properties.get('status')
+
+    @property
+    def started(self):
+        """Time when the change set was started.
+
+        :rtype: ``datetime.datetime`` or ``NoneType``
+        :returns: Time, as set by the back-end, or None.
+        """
+        stamp = self._properties.get('startTime')
+        if stamp is not None:
+            return datetime.datetime.strptime(stamp, _RFC3339_MICROS).replace(
```

Comment:
Realizing we've got this in `storage.bucket`, `storage.blob` (twice) and `pubsub.message` too. We should consolidate into `gcloud._helpers`.

You can do it in this PR or if you want to punt, LMK and I'll file and issue / send a PR.